### PR TITLE
Feature/ees 310 footnotes ui

### PR DESCRIPTION
**Pull request by bennettstuart**
Create Footnotes in Admin

### CODE REVIEW

File: src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        });
+      });
+    });
+  });
+
+  const getItem = (
+    ids: number[],
+    itemType: 'subject' | 'indicator' | 'filter',
+  ) => {
+    //@ts-ignore
```
Comment (negatron99) [inline src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts]: You could get rid of this by changing the parameter type to `(number | undefined)[]` and casting to `number[]` later maybe?

File: src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    if (ids.indexOf(undefined) + 1 || ids.length === 0) {
```
Comment (negatron99) [inline src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts]: Would it be better to use the `Array.includes` function ?

Comment (bennettstuart) [reply src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts]: Yep, good point :)

File: src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts (hunk 1 above)
Comment (bennettstuart) [reply src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts]: I'd like to keep it as is-
Logically the array should only ever have numbers - depending on the fact there isn't any bugs in both the backend and frontend code

The check for `undefined` is a fail safe for the mapping functions above which is based on the footnoteMeta that's returned on the backend. 

So, in the instance of a bug - it won't error, just send UI to alert the user of an error

Comment (negatron99) [reply src/explore-education-statistics-admin/src/services/release/edit-release/footnotes/util.ts]: I'd like to start removing the `ts-ignore` as much as possible from the code tbh, I know it's an annoying process, but it's typescript for a reason.

**Pull request merged**